public/scripts/index.js

```diff
@@ -0,0 +1,171 @@
+import { calcNewCellState, makeMatrix, calcCellCoordinates, calcClickCoordinates, getIntermediateFrame, drawBoard, getPreset, draw} from './engine.js';
+
+let currentFrame;
+let intermediateFrame; //For holding the neighbour count
+let nextFrame; 
+let spanMouseX = document.getElementById('mouse-x'); 
```

Comment:
All the calls to `getElementById` should be moved inside `init` func becayse it's guaranteed to run after the DOM finished loading